### DISCUSSION TITLE
[DISCUSSION]: Composition API, Typescript, and rewrites.

### DISCUSSION THREAD

**Discussion by LeaderbotX400** [primefaces/community#2650] [PrimeVue]
The purpose of this discussion, is to discuss the feasibility and potential desire to migrate to Vue's composition API using the [script setup](https://vuejs.org/api/sfc-script-setup.html) syntax. As well as migration to  a more modular composable based system as component extensions aren't a thing within the composition API.

While this will be a major task, effectively rewriting every component. I believe it will be worth the trade off. Moving to the composition API forces components to be a little more self-contained but allows for more effective use of things like Vue's support for typed [generics](https://vuejs.org/api/sfc-script-setup.html#generics) in things like `defineModel`, `defineProps`, and `defineEmits`.

The Above will be a much needed DX improvement on components like `Timeline`, `DataTable`, etc.

**Comment by elsayed1** [comment]
intersted
